src/gpio.rs

```diff
@@ -1,4 +1,39 @@
 //! GPIO and pin configuration
+//!
+//! Interface for the input/output pins.
+//!
+//! `Gpio1` through `Gpio39` represent the *physical* pins of the ESP chip.
+//!
+//! *Logical* pins (compatible with `embedded_hal::digital::InputPin`/`OutputPin`)
+//! are implemented through `PinDriver`.
+//!
+//! The ESP architecture has a I/O multiplexer, which means that (almost) any
+//! physical pin can be used for any logical function (i.e. GPIO, I2C, SPI, ADC, etc).
+//! Even though it's possible to use a pin for several functions at once, this
```

Comment:
Is that really possible? Even if it is possible, it is not possible with the safe API of `esp-idf-hal` for sure, as the pin peripheral (what you call above "physical" pins) can only be used by one driver at a time.

Reply:
From what I'm reading at https://hackaday.io/page/21183-hacking-esp32-multiplexing-i2c , it seems that it's possible to use the same pin peripheral ("physical") as the CLK pin in two different I2C interfaces at once.

Which means one of the I2C buses is working as expected, and the other one is receiving all ones (or all `0xFF`s) on the SDA pin, which is innocuous enough.

Reply:
Interesting. How would you do that with the ESP IDF C API?